src/test/scala/edu/ie3/simona/agent/em/EmAgentIT.scala

```diff
@@ -0,0 +1,433 @@
+            hpInputModel.getNode.getGeoPosition.getX,
+          )
+        )
+
+        heatPumpAgent ! RegistrationSuccessfulMessage(
+          weatherService.ref.toClassic,
+          Some(0L),
+        )
+
+        scheduler.expectMessage(Completion(heatPumpAgent))
+
+        val weatherDependentAgents = Seq(pvAgent, heatPumpAgent)
+
+        /* TICK 0
+         LOAD: 0.000269 MW
+         PV:  -0.005685 MW
+         Heat pump: off, can be turned on or stay off
+         -> set point ~3.5 kW (bigger than 50 % rated apparent power): turned on
+         -> remaining -0.000566 MW
+         */
+
+        emAgentActivation ! Activation(0)
+
+        weatherDependentAgents.foreach {
+          _ ! ProvideWeatherMessage(
+            0,
+            weatherService.ref.toClassic,
+            WeatherData(
+              WattsPerSquareMeter(400d),
+              WattsPerSquareMeter(200d),
+              Celsius(0d),
+              MetersPerSecond(0d),
+            ),
+            Some(7200),
+          )
+        }
+
+        resultListener.expectMessageType[ParticipantResultEvent] match {
+          case ParticipantResultEvent(emResult: EmResult) =>
+            emResult.getInputModel shouldBe emInput.getUuid
+            emResult.getTime shouldBe 0.toDateTime
+            emResult.getP should equalWithTolerance(
+              (-0.000566087824).asMegaWatt
+            )
+            emResult.getQ should equalWithTolerance(0.001073120041.asMegaVar)
+        }
+
+        scheduler.expectMessage(Completion(emAgentActivation, Some(7200)))
+
+        /* TICK 7200
+         LOAD: 0.000269 MW (unchanged)
+         PV:  -0.003797 MW
+         Heat pump: running (turned on from last request), can be turned on and stay off
+         -> set point ~3.5 kW (bigger than 50 % rated apparent power): stays turned on with unchanged state
+         -> remaining 0 MW
+         */
+
+        emAgentActivation ! Activation(7200)
+
+        weatherDependentAgents.foreach {
+          _ ! ProvideWeatherMessage(
+            7200,
+            weatherService.ref.toClassic,
+            WeatherData(
+              WattsPerSquareMeter(300d),
+              WattsPerSquareMeter(500d),
+              Celsius(0d),
+              MetersPerSecond(0d),
+            ),
+            Some(14400),
+          )
+        }
+
+        resultListener.expectMessageType[ParticipantResultEvent] match {
+          case ParticipantResultEvent(emResult: EmResult) =>
+            emResult.getInputModel shouldBe emInput.getUuid
+            emResult.getTime shouldBe 7200.toDateTime
+            emResult.getP should equalWithTolerance(0.00132184544484.asMegaWatt)
+            emResult.getQ should equalWithTolerance(0.001073120041.asMegaVar)
+        }
+
+        scheduler.expectMessage(Completion(emAgentActivation, Some(14400)))
+
+        /* TICK 14400
+         LOAD: 0.000269 MW (unchanged)
+         PV:  -0.000066 MW
+         Heat pump: Is not running, can still run or be turned off
```

Comment:
"Is not running, can still run", was running the step before. At least the comment need to be fixed. Not sure about the values.

Reply:
True, comment not 100% correct